src/map/status.cpp

```diff
@@ -9266,10 +9268,10 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 		for (const auto &it : scdb->fail) {
 			// Don't let OPT1 that have RemoveOnDamaged start a new effect in the same attack.
 			if (sc->lastEffect == it) {
-				sc->lastEffect = SC_NONE;
+				if (sc->lastEffectTimer == INVALID_TIMER)
+					sc->lastEffectTimer = add_timer(gettick() + 10, status_clear_lastEffect_timer, bl->id, 0);
```

Comment:
Wrong place to start the timer, you want to start the timer the moment you set lastEffect.